editor/editor_node.cpp

```diff
@@ -1935,6 +1935,21 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 			opening_prev = true;
 			open_request(previous_scenes.back()->get());
 
+		} break;
+		case FILE_CLOSE_OTHERS:
+		case FILE_CLOSE_RIGHT:
+		case FILE_CLOSE_ALL: {
+			if (editor_data.get_edited_scene_count() > 1 && (current_option != FILE_CLOSE_RIGHT || editor_data.get_edited_scene() < editor_data.get_edited_scene_count() - 1)) {
+				int next_tab = editor_data.get_edited_scene() + 1;
+				next_tab %= editor_data.get_edited_scene_count();
+				_scene_tab_closed(next_tab, current_option);
+			} else {
+				if (current_option != FILE_CLOSE_ALL)
+					current_option = -1;
+				else
+					_scene_tab_closed(editor_data.get_edited_scene());
```

Comment:
Looks like you're maybe missing the second argument (option) in the call to _scene_tab_closed() here?

Reply:
Nope, this happens when you want to close all tabs and there's only one tab left. It closes the tab with standard SCENE_TAB_CLOSE to finalize the closing.

Reply:
Ah, gotcha :)